functions/go/set-labels/README.md

```diff
@@ -4,29 +4,45 @@
 
 <!--mdtogo:Short-->
 
-Add a list of labels to all resources.
+The `set-labels` function adds a list of [labels] to all resources. It's a
+common practice to add a set of labels for all the resource in a package.
+Kubernetes has some [recommended labels].
+
+For example, labels can be used in the following scenarios:
+
+- Identify the KRM resources by querying their labels.
+- Set labels for all resources within a package (e.g. environment=staging).
```

Comment:
We can point to kubernetes docs at the end of the overview.

You can learn more about labels [here](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/).